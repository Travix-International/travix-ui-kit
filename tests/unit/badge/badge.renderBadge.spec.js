import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Badge from '../../../components/badge/badge';

describe('Badge', () => {
  describe('#renderBadge()', () => {
    it('should render arrow-left', () => {
      const wrapper = shallow(
        <Badge arrow position="right" title="Badge title">
          Badge Content
        </Badge>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render arrow-right', () => {
      const wrapper = shallow(
        <Badge arrow position="left" title="Badge title" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should ignore position if no content provided', () => {
      const wrapper = shallow(
        <Badge position="right" title="Badge title" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
