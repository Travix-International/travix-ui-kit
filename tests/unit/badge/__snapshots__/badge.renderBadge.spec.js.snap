// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge #renderBadge() should ignore position if no content provided 1`] = `
<div
  className="ui-badge-badge"
>
  Badge title
</div>
`;

exports[`Badge #renderBadge() should render arrow-left 1`] = `
<div
  className="ui-badge"
>
  Badge Content
  <div
    className="ui-badge-badge ui-badge-badge_arrow-left ui-badge-badge_right"
  >
    Badge title
    <span
      className="ui-badge-badge-arrow"
    />
  </div>
</div>
`;

exports[`Badge #renderBadge() should render arrow-right 1`] = `
<div
  className="ui-badge-badge ui-badge-badge_arrow-right"
>
  Badge title
  <span
    className="ui-badge-badge-arrow"
  />
</div>
`;
